core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -199,9 +203,20 @@ impl BlocksWeb3Dal<'_, '_> {
             transactions_root: H256::zero(),
             receipts_root: H256::zero(),
             number: Some(U64::from(row.number)),
-            gas_used: U256::zero(),
-            gas_limit: U256::zero(),
-            base_fee_per_gas: None,
+            gas_used: {
+                if let (Some(gas_limit), Some(refunded_gas)) =
+                    (row.block_gas_limit, row.refunded_gas)
+                {
+                    bigdecimal_to_u256(gas_limit.into()) - U256::from(refunded_gas as u64)
+                } else {
+                    U256::zero()
+                }
+            },
```

Comment:
This is incorrect, you should do `transaction_gas_limit - refunded_gas`.

Also, you should take into account that you now join `transactions` table, number of rows returned will be equal to number of transactions, not number of blocks, so doing `let blocks = rows.into_iter().map(..` is incorrect, check how it's done in `get_api_block` with `fold` and reuse. Alternatively, you can avoid joining and select aggregated `transaction_gas_limit - refunded_gas` in SQL with subquery